### ISSUE TITLE
AMD GPU support in PyTorch

### ISSUE THREAD

**Issue by hpigula** [pytorch/pytorch#10657]
PyTorch version: 0.4.1.post2
Is debug build: No
CUDA used to build PyTorch: None

OS: Arch Linux
GCC version: (GCC) 8.2.0
CMake version: version 3.11.4

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] Could not collect
[conda] Could not collect

Hello I'm running latest PyTorch on my laptop with AMD [A10-9600p](https://www.amd.com/en/support/apu/amd-series-processors/amd-a10-series-apu-for-laptops/7th-gen-a10-9600p-apu) and it's iGPU that I wish to use in my projects but am not sure if it works and if yes how to set it to use iGPU but have no CUDA support on both Linux(Arch with Antergos) and Win10 Pro Insider so it would be nice to have support for something that AMD supports.

**Comment by Jorghi12** [comment]
Hello,

**Disclaimer:** PyTorch AMD is still in development, so full test coverage isn't provided just yet.

PyTorch AMD runs on top of the Radeon Open Compute Stack (ROCm), which you can install from over here: https://rocm.github.io/install.html.

Once installed, then run the cuda->hip transpiler & build PyTorch.

- python tools/amd_build/build_pytorch_amd.py
- USE_ROCM=1 python setup.py install

For the list of AMD GPUs the ROCm stack supports, see https://github.com/RadeonOpenCompute/ROCm.

**Comment by hpigula** [comment]
Nah for now according to their site neither my cpu nor my gpu are supported so as fro next few months I'll have to do with CPU only PT though I still am trying to learn Deep Learning and other AI.

Thanks for help :)

**Comment by variadico** [comment]
> run the cuda->hip transpiler
How do you do that?

**Comment by hyperfraise** [comment]
Have you checked this ? https://rocm-documentation.readthedocs.io/en/latest/Deep_learning/Deep-learning.html

**Comment by mirh** [comment]
It is of my understanding that Excavator v2 cpu *has* PCIe 3.0 (and corresponding atomics then). 
As for the iGPU, support is likely waiting on XNACK https://github.com/RadeonOpenCompute/hcc/issues/879#ref-issue-332614110

**Comment by javiqm12** [comment]
Pytorch on AMD graphic cards is long waited capability and highly appreciated. Wishing hardly to get this soon, thanks.

**Comment by auvipy** [comment]
I am going to try this. wish me luck!

**Comment by kir486680** [comment]
> I am going to try this. wish me luck!

omg please do it

**Comment by acxz** [comment]
While PyTorch does not provide official pip package for torch-rocm, on Arch Linux you can install the `python-pytorch-rocm` package to use pytorch with the AMD GPU. See: https://github.com/rocm-arch/python-pytorch-rocm

**Comment by t-vi** [comment]
Here is also my recipe for compiling on ROCm https://lernapparat.de/pytorch-rocm/
Though this might be less needed in a bit...

**Comment by Filco306** [comment]
Is it possible to use one NVIDIA and one AMD graphics card together? Has anybody done this, or does anybody know of any resources showing how to do this?

**Comment by t-vi** [comment]
@Filco306 You can have them in the same machine, but not in the same PyTorch. The reason is mainly the technical decision of having the HIP version patch the CUDA bits into HIP bits. For many, (most?) things, this goes into new .hip files, so these would likely be OK, but the headers for the CUDA bookkeeping itself are replacing CUDA with HIP versions (e.g. the CUDAContext) . If this were abstracted instead, on could have that, e.g. Apache TVM can run both in parallel.

What's your use case?

**Comment by Filco306** [comment]
Thank you @t-vi for your reply! Well, basically, I have an AMD graphics card built into my MacBook Pro. However, I will through my work have an additional GPU attachable to run for smaller experiments. This will be a way better one than my AMD, but if I would be able to use both would of course be beneficial.

**Comment by t-vi** [comment]
@Filco306 So quite likely, switching between PyTorch installations would work then. However, note that ROCm supports only [selected AMD graphics cards](https://github.com/RadeonOpenCompute/ROCm#supported-gpus) and is for Linux (but the implications are strictly offtopic here and there are issues over at [the ROCm github project](https://github.com/RadeonOpenCompute/ROCm/issues) discussing them).

**Comment by Filco306** [comment]
@t-vi yes, that seems like the approach. And yes, I do know that I need Linux, so will probably partition my disk and install Ubuntu as well. Thanks!

**Comment by daniele777** [comment]
Guys help to understand  i follow your project and this 

https://github.com/LearnedVector/A-Hackers-AI-Voice-Assistant

I setting up ambient and all ok when i train neural network based pytorch  i use this Command :

python3 train.py --sample_rate 8000 --epochs 100 --batch_size 32 --eval_batch_size 32 --lr 0.0001 --model_name ./new_wakeword_v0 --save_checkpoint_path neuralnet/checkpoints --train_data_json data/json/train.json --test_data_json data/json/test.json --num_workers 4 --hidden_size 64

 I have slow process can i add at command like -gpu amd ? . Can explain me please if possibile to use amd video card .?
and add command --use gpu or somethink like that???

BestRegards 


Attachments area

**Comment by hongxiayang** [comment]
Please check the updated documentation on this front:
Latest documentation of ROCm: https://rocm.docs.amd.com/en/latest/
Latest documentation of ROCm support on GPUs: https://rocm.docs.amd.com/en/latest/release/gpu_os_support.html#linux-supported-gpus

We will close this issue as this is quite old and outdated.
